api/src/modules/impact/impact.service.ts

```diff
@@ -163,4 +173,101 @@ export class ImpactService {
     });
     return purchasedTonnes;
   }
+
+  /**
+   *
+   * @description Get a tree of Materials imported by a User
+   */
+
+  async getMaterialTreeForImpact(): Promise<Material[]> {
+    const materialIdsFromSourcingLocations: string[] =
+      await this.sourcingLocationService.getMaterialIdsAndParentIds();
+    const allMaterials: Material[] =
+      await this.materialsService.findAllUnpaginated();
+    const materialLineage: Material[] = this.getAncestry<Material>(
+      allMaterials,
+      materialIdsFromSourcingLocations,
+    );
+    return this.materialsService.serialize(
+      this.buildTree<Material>(materialLineage, null),
+    );
+  }
+
+  /**
+   *
+   * @description Get a tree of AdminRegions imported by a User
+   */
+
+  async getAdminRegionTreeForImpact(): Promise<any> {
+    const adminRegionIdsFromSourcingLocations: string[] =
+      await this.sourcingLocationService.getAdminRegionIdsAndParentIds();
+    const allAdminRegions: AdminRegion[] =
+      await this.adminRegionsService.findAllUnpaginated();
+    const adminRegionLineage: AdminRegion[] = this.getAncestry<AdminRegion>(
+      allAdminRegions,
+      adminRegionIdsFromSourcingLocations,
+    );
+
+    return this.adminRegionsService.serialize(
+      this.buildTree<AdminRegion>(adminRegionLineage, null),
+    );
+  }
+
+  /**
+   *
+   * @description Get a tree of Suppliers imported by a User
+   */
+
+  async getSupplierTreeForImpact(): Promise<any> {
+    const supplierIdsFromSourcingLocations: string[] =
+      await this.sourcingLocationService.getSupplierIdsAndParentIds();
+    const allSuppliers: Supplier[] =
+      await this.suppliersService.findAllUnpaginated();
+    const supplierLineage: Supplier[] = this.getAncestry<Supplier>(
+      allSuppliers,
+      supplierIdsFromSourcingLocations,
+    );
+    return this.suppliersService.serialize(
+      this.buildTree<Supplier>(supplierLineage, null),
+    );
+  }
+
+  getAncestry<T extends { id: string; parentId?: string; children?: T[] }>(
```

Comment:
please document what this does. I tried looking at it, but my brain started to hurt. I need to re-review this once docs are in place